packages/core/src/utils/assertion.ts

```diff
@@ -80,6 +80,8 @@ export const isRootKey = (specKey: string, target: string) => {
   return specKey === target;
 };
 
+const LOCALHOST_REGEX = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i;
+
 export const isUrl = (str: string) => {
-  return validatorIsUrl(str, { require_tld: false });
+  return validatorIsUrl(str) || LOCALHOST_REGEX.test(str);
```

Comment:
Why using the validator if the regex handles all the cases?

Reply:
That is a great question I am more than happy to get rid of the validator? But that is a question for @anymaniax ?

Reply:
The package validator is only brought it for this single check so it would allow us to remove another dependency.

Reply:
Updated my solution using Snyk's suggestion: https://snyk.io/blog/secure-javascript-url-validation/